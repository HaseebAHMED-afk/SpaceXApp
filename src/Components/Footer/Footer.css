.footer{
    background-color: rgb(0, 0, 0);
    height: 100px;
    color: rgb(175, 175, 175);
    overflow: hidden;
    text-align: center
}
p{
    margin: 50px;
    font-family: 'Sawarabi Gothic', sans-serif;
}