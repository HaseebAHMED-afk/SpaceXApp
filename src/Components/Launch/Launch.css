.launch-heading-section{
    text-align: center;
    color: white;
    margin: 25px auto;
}
.main-heading{
    font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
}
.main-para{
    font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
}

.launch-deck{
    display: flex; 
    flex-wrap: wrap;
    justify-content: center;
}
.launch-card{
    background-color: white;
    width: 25%;
    height: 120px;
    margin: 25px 15px;
    font-family: 'Ubuntu', sans-serif;
    overflow: hidden;
    border-radius: 3px;
}
.launch-name{
   margin: 10px; 
   font-size: 100%;
}

.launch-year, .launch-success{
    margin: 10px;
    display: inline;
    font-size: 100%;
}



@media only screen and (max-width:700px) {
    .launch-card{
        width: 45%;
    }
    .launch-success{
        display: block;
    }
}

@media only screen and (max-width: 350px){
    .launch-card{
        width: 100%
    }
    .launch-success{
        display: block;
    }
}