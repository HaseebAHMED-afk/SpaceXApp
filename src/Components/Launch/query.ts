import gql from 'graphql-tag'

export const QUERY_LAUNCH = gql `
query Launches{
    launches {
      mission_name
      launch_success
      launch_year
    }
  }
`;