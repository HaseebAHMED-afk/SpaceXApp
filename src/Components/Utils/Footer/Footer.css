.footer{
    background-color: rgb(255, 255, 255);
    height: 100px;
    color: rgb(0, 0, 0);
    overflow: hidden;
    text-align: center;
    padding-top: 50px;
}
.footer-para{
    margin: 0;
    font-family: 'Sawarabi Gothic', sans-serif;
    color: black;
}