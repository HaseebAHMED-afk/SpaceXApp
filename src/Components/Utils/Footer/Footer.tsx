import React from 'react'
import "./Footer.css"

const Footer = () => {
    return (
        <div className="footer" >
            <p className="footer-para">Haseeb Ahmed Enterpriseâ„¢</p>
        </div>
    )
}

export default Footer
