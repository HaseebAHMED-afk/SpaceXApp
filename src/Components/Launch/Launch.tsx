import React from 'react'
import { LaunchesQuery } from  '../../generated/graphql'
import './Launch.css'

interface Props {
    data: LaunchesQuery,
}

const Launch: React.FC<Props> = ({data}) => {
    return (
        <div className="Launches">
            <h3>SpaceX Launches</h3>
            <ol className="launches-ol">
                {!!data.launches && data.launches.map(
                    (launch,i) => !!launch && 
                     (
                        <li key={i} className="launch-item">
                            <h1>{launch.mission_name}</h1>
                            <p>{launch.launch_year}</p>
                            <p>{launch.launch_success}</p>
                        </li>
                    )
                )}
            </ol>
        </div>
    )
}

export default Launch 
