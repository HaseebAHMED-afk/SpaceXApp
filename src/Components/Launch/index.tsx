import React from 'react'
import {useLaunchesQuery} from '../../generated/graphql'
import Launch from './Launch'

const LaunchContainer = () =>{
    const {data,error,loading} = useLaunchesQuery();

    if(loading){
        return <div>Data is Loading</div>
    }

    if(error || !data){
        return <div>Error</div>
    }

    if(data){
        return <Launch data={data} /> 
    }

}

export default LaunchContainer