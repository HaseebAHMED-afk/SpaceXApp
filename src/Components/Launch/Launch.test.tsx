import React from 'react'
import {shallow} from 'enzyme'
import Launch from './Launch'
import Footer from '../Utils/Footer/Footer'

describe('renders the Launch' ,() =>{

    let container : any;

    beforeEach(() => container = shallow(<Launch  />))

    it('renders the components' , () => {
        expect(container).toMatchSnapshot();
    })

    it('renders the div' , () => {
        expect(container.find('div').length).toBeGreaterThanOrEqual(1)
    })

    it('renders the footer' ,() =>{
        expect(container.containsMatchingElement(<Footer />)).toEqual(true)
    } )
})