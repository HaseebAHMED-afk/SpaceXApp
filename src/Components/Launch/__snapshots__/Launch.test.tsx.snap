// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Launch renders the components 1`] = `
<div
  className="launches"
>
  <div
    className="launch-heading-section"
  >
    <h1
      className="main-heading"
    >
      SpaceX Launches
    </h1>
    <p
      className="main-para"
    >
      See the launches that describe SpaceX's journey to the great beyond
    </p>
  </div>
  <div
    className="launch-deck"
  />
  <Footer />
</div>
`;
